api/document/js/parse-doc.ts

```diff
@@ -25,6 +43,13 @@ export function convertContent(content, marks: Mark[]): Node[] {
   return flatten(nestedChildNodes);
 }
 
+function mapChildren(node: ApiNode): Node[] {
+  return (node.children || []).filter((child) => {
+    // We're going to strip out footnotes, as we're using the
+    // footnote node embedded in the citation instead.
+    return (child.node_type !== 'footnote');
```

Comment:
This raises a good point -- we're potentially losing data if the footnote isn't referenced anywhere, which we know is a frequent occurrence from out pdf parser's output. I think we'll need a way to catch those footnotes -- what do you think about keeping
```js
policy: node => factory.policy((node.children || []).map(parseDoc))
```
(and adding commentary explaining what's going on)? I'm pretty sure we put uncited footnotes as children of the root `policy`, so I think that'd remove the risk of lost data.

Reply:
Oh blarg, right.  Um, so when @rtwell and I were talking, we actually discussed changing things on the PDF importer side, rather than on the editor side--e.g. ensuring the PDF importer (and the schema validator that we have yet to write on the backend) only made footnotes that had citations, and any footnotes that lacked citations would simply be converted into paragraphs (potentially with some commentary from the importer mentioning that they thought it was a footnote, but couldn't find the citation).

Umm let me think about this a bit and get back 2u.

Reply:
So, I have some concerns about keeping uncited footnotes around the PM schema as "not-footnotes"--specifically, I want to eventually add logic for adding and removing footnotes with auto-numbering, but if that logic needs to account for uncited footnotes, it'll complicate the logic a lot.

(Also, I'm unclear on where the line of code you quoted is supposed to go, or how it'd be used...)

What if we, at the API level (although I guess we could do it as a pre-processing step in the editor too), actually output the document ensuring that all footnotes had citations, _inserting_ citations in their most appropriate locations if they didn't exist?  e.g., if a document had footnotes+citations for 1 and 3, but only a footnote w/o a citation for 2, it could simply find the citation for 1 and add a citation for 2 immediately after it.

Reply:
I think that's an interesting strategy, though I'd put that at the pdf-to-database layer. Ideally, we're removing the `footnote` (api) node type altogether and replacing it with those inline versions; getting there will take some time...

Reply:
Cool, that sounds like a good idea; I can file a PR for it later. For now I understand your one-line solution, now that you explained it to me over Slack, so I will bust that up!